README
 ************** MCFM - version 8.1 ****************
 *                                                *
 * MCFM, v8.1                      December 2017  *
 *                                                *
 * Authors: John Campbell, Keith Ellis,           *
 *          Walter Giele, Tobias Neumann,         *
 *          Ciaran Williams                       *
 *         (johnmc@fnal.gov, ellis@fnal.gov,      *
 *          giele@fnal.gov,tneumann@fnal.gov,     *
 *          ciaranwi@buffalo.edu)                 *
 *                                                *
 * For details see:                               *
 *                                                *
 *  Color singlet production at NNLO in MCFM      *
 *   R. Boughezal, J. Campbell, R.K. Ellis,       *
 *    C. Focke, W. Giele, X. Liu, F. Petriello,   *
 *    C. Williams,  arXiv:1605.08011              *
 *    (overview of NNLO implementation in MCFM)   *
 *                                                *
 *  A study of weak corrections to Drell-Yan,     *
 *   top-quark pair and di-jet production at      *
 *    high energies with MCFM                     *
 *   J. M. Campbell, D. Wackeroth, J. Zhou,       *
 *   arXiv:1608.03356 (weak corrections in MCFM)  *
 *                                                *
 *  arXiv:1708.02925 (Zgamma at NNLO)             *
 *  arXiv:1609.00367 (NLO H+jet with mass effects)*
 *  arXiv:1603.02663 (diphotons at NNLO)          *
 *  arXiv:1601.00658 (VH at NNLO)                 *
 *                                                *
 *  arXiv:1502.02990 (VBF and VBS Higgs)          *
 *  arXiv:1403.2641  (Triphoton production)       *
 *  arXiv:1312.1628  (gg->WW, Higgs interference) *
 *  arXiv:1311.3589  (gg->ZZ, Higgs interference) *
 *  Phys.Rev.D87:114006, arXiv:1302.3856          *
 *  (tZ, tH -- with R. Rontsch)                   *
 *  arXiv:1211.6390 (DM, P. Fox and C. Williams)  *
 *  JHEP 1211:162 (2012), arXiv:1208.0566         *
 *  (Z+gam+jet,Z+gam+gam -- with H. Hartanto)     *
 *  arXiv:1204.1513 (top production+decay)        *
 *  JHEP 1207:052 (2012), arXiv:1204.5678 (ttW)   *
 *  JHEP 1110:005 (2011), arXiv:1107.5569         *
 *         (gg->WW,Higgs intference)              *
 *  JHEP 1107:018 (2011), arXiv:1105.0020         *
 *         (diboson update)                       *
 *  JHEP 1103:027 (2011), arXiv:1011.6647         *
 *         (Wbb for mb>0, with S. Badger)         *
 *  Phys.Rev.D81:074023, arXiv:1001.4495 (H+2jet) *
 *                                                *
 *  P.R.L. 102:142001, arXiv:0903.0005 [hep-ph]   *
 *    (t-channel single top + explicit b,         *
 *      JC, R.Frederix, F.Maltoni, F.Tramontano)  *
 *  N.P.B 726:109(2005), hep-ph/0506289 (W+t)     *
 *  Phys.Rev.D70:094012, hep-ph/0408158 (Sngl Top)*
 *       (with Francesco Tramontano)              *
 *                                                *
 *  Phys.Rev.D65:113007, hep-ph/0202176 (W,Z+2j)  *
 *  Phys.Rev.D62:114012, hep-ph/0006304 (W,Z+bb)  *
 *  Phys.Rev.D60:113006, hep-ph/9905386 (diboson) *
 *                                                *
 * On the web:  http://mcfm.fnal.gov/             *
 **************************************************
 *                                                *
 * To compile:           ./Install                *
 *   and then:           make                     *
 *                                                *
 * To run:               cd Bin                   *
 *                       ./mcfm                   *
 *                                                *
 * To change the process, collider, cuts, etc.    *
 *  edit the file Bin/input.DAT                   *
 *                                                *
 * Further help is in the Doc/ directory.         *
 *                                                *
 **************************************************


CHANGELOG.md
# Changelog

## [8.1] - November 2017
### Added
- Electroweak one-loop corrections for $Z$, $t\bar t$ and di-jet production [1].
- $Z\gamma$ process at NNLO including anomalous couplings [2].
- $Z\gamma$ decay for H+jet production [2].
- $H$+2jet process with a finite top-quark mass and $H$+jet with finite
  top-quark mass effects [1]. This adds a new parameter [mtex] to the
  the input file, which specifies to which order in 1/mt^k (k=0,2,4) the finite
  part of the virtual corrections are computed.
- Support for random seeds by setting the seed value (previously [ij]) to 0.
- Support for boosted (as opposed to hadronic) definition of jettiness, which
  is now also used as default.
- Support for $p_T$ and rapidity ranges for most cuts in the input file.
- Native implementation of two PDF sets containing photons: mrstqed and
  CT14qed.
- EXPERIMENTAL: Integration routine that adaptively selects the cross section
  contribution with the largest integration uncertainty. It also adapts
  integration calls in the warmup phase and then continuously increases them to
  not run into grid bias problems. This ingration can be enabled by setting
  newIntegration to .true. in Need/mcfmmain.f. After each iteration a snapshot is
  saved for resumption. When readin is set to .true. in the gridinfo_logic
  common block, the integration will be resumed from any stage. Currently this
  integration mode does not stop the integration, but the user can at any point
  abort manually when the desired precision has been reached. It has only been
  thoroughly tested with $Z\gamma$ and $H+$jet. Other processes might need
  tweaks or specific initializations.

### Changed
- Fixed FROOT implementation.
- Upgraded QCDLoop to QCDLoop 2.
- Using new random number generator (Mersenne Twister) from libstdc++ (C++11).
- Fixed bug concerning integration uncertainty of histogram bins.

### Removed
- Removed flags virtonly,realonly,vanillafiles in input.DAT, as well as broken
  parameters for resuming integrations.
- Removed support for PDFLIB.
- Removed noomp version. If just a single thread is wanted, please
  set OMP_NUM_THREADS to 1.

[1] Campbell, Wackeroth, Zhou https://arxiv.org/abs/1608.03356
[2] Neumann, Williams https://arxiv.org/abs/1609.00367
[3] Campbell, Neumann, Williams https://arxiv.org/abs/1708.02925


src/Need/banner.f
      block data codeversion_data
      
      include 'codeversion.f'
      data codeversion/'8.1'/      
      data      prelim/.false./      ! if true, print warning message
      end

      subroutine banner
      implicit none
      include 'types.f'
************************************************************************
*  Set the version number of MCFM and write out the banner heading     *
************************************************************************
      
      include 'codeversion.f'
      character*50 line
      integer:: vlength,lenocc

      line='**************************************************'
      vlength=lenocc(codeversion)
      vlength=vlength+17   
      line(25-vlength/2:24+(vlength+1)/2)=
     & ' MCFM - version '//codeversion//' '

      write(6,*) line

c--- warning message, if necessary
      if (prelim) then
        write(6,*) '*                                                *'
        write(6,*) '*           PRELIMINARY VERSION                  *'
        write(6,*) '*                                                *'
        write(6,*) '*  NOTE: This is a private release of the MCFM   *'
        write(6,*) '*  code that has not yet been made public on the *'
        write(6,*) '*  usual website. As such:                       *'
        write(6,*) '*                                                *'
        write(6,*) '*   + Please do not redistribute without the     *'
        write(6,*) '*     knowledge of the authors;                  *'
        write(6,*) '*                                                *'
        write(6,*) '*   + Please notify the authors of any bugs      *'
        write(6,*) '*     or problems so that they can be corrected  *'
        write(6,*) '*     before the next official release.          *'
        write(6,*) '*                                                *'
        write(6,*) line
      endif
     

      write(6,*) '*                                                *'
      write(6,*) '* MCFM, v'//codeversion//
     &                          '                   December 2017  *'
      write(6,*) '*                                                *'
      write(6,*) '* Authors: John Campbell, Keith Ellis,           *'
      write(6,*) '*          Walter Giele, Tobias Neumann,         *'
      write(6,*) '*          Ciaran Williams                       *'
      write(6,*) '*         (johnmc@fnal.gov, ellis@fnal.gov,      *'
      write(6,*) '*          giele@fnal.gov,tneumann@fnal.gov,     *'
      write(6,*) '*          ciaranwi@buffalo.edu)                 *'
      write(6,*) '*                                                *'
      write(6,*) '* For details see:                               *'
      write(6,*) '*                                                *'
      write(6,*) '*  Color singlet production at NNLO in MCFM      *'
      write(6,*) '*   R. Boughezal, J. Campbell, R.K. Ellis,       *'
      write(6,*) '*    C. Focke, W. Giele, X. Liu, F. Petriello,   *'
      write(6,*) '*    C. Williams,  arXiv:1605.08011              *'
      write(6,*) '*    (overview of NNLO implementation in MCFM)   *'
      write(6,*) '*                                                *'
      write(6,*) '*  A study of weak corrections to Drell-Yan,     *'
      write(6,*) '*   top-quark pair and di-jet production at      *'
      write(6,*) '*    high energies with MCFM                     *'
      write(6,*) '*   J. M. Campbell, D. Wackeroth, J. Zhou,       *'
      write(6,*) '*   arXiv:1608.03356 (weak corrections in MCFM)  *'
      write(6,*) '*                                                *'
      write(6,*) '*  arXiv:1708.02925 (Zgamma at NNLO)             *'
      write(6,*) '*  arXiv:1609.00367 (NLO H+jet with mass effects)*'
      write(6,*) '*  arXiv:1603.02663 (diphotons at NNLO)          *'
      write(6,*) '*  arXiv:1601.00658 (VH at NNLO)                 *'
      write(6,*) '*                                                *'
      write(6,*) '*  arXiv:1502.02990 (VBF and VBS Higgs)          *'
      write(6,*) '*  arXiv:1403.2641  (Triphoton production)       *'
      write(6,*) '*  arXiv:1312.1628  (gg->WW, Higgs interference) *'
      write(6,*) '*  arXiv:1311.3589  (gg->ZZ, Higgs interference) *'
      write(6,*) '*  Phys.Rev.D87:114006, arXiv:1302.3856          *'
      write(6,*) '*  (tZ, tH -- with R. Rontsch)                   *'
      write(6,*) '*  arXiv:1211.6390 (DM, P. Fox and C. Williams)  *'
      write(6,*) '*  JHEP 1211:162 (2012), arXiv:1208.0566         *'
      write(6,*) '*  (Z+gam+jet,Z+gam+gam -- with H. Hartanto)     *'
      write(6,*) '*  arXiv:1204.1513 (top production+decay)        *'
      write(6,*) '*  JHEP 1207:052 (2012), arXiv:1204.5678 (ttW)   *'
      write(6,*) '*  JHEP 1110:005 (2011), arXiv:1107.5569         *'
      write(6,*) '*         (gg->WW,Higgs intference)              *'
      write(6,*) '*  JHEP 1107:018 (2011), arXiv:1105.0020         *'
      write(6,*) '*         (diboson update)                       *'
      write(6,*) '*  JHEP 1103:027 (2011), arXiv:1011.6647         *'
      write(6,*) '*         (Wbb for mb>0, with S. Badger)         *'
      write(6,*) '*  Phys.Rev.D81:074023, arXiv:1001.4495 (H+2jet) *'
      write(6,*) '*                                                *'
      write(6,*) '*  P.R.L. 102:142001, arXiv:0903.0005 [hep-ph]   *'
      write(6,*) '*    (t-channel single top + explicit b,         *'
      write(6,*) '*      JC, R.Frederix, F.Maltoni, F.Tramontano)  *'
      write(6,*) '*  N.P.B 726:109(2005), hep-ph/0506289 (W+t)     *'
      write(6,*) '*  Phys.Rev.D70:094012, hep-ph/0408158 (Sngl Top)*'
      write(6,*) '*       (with Francesco Tramontano)              *'
      write(6,*) '*                                                *'
      write(6,*) '*  Phys.Rev.D65:113007, hep-ph/0202176 (W,Z+2j)  *'
      write(6,*) '*  Phys.Rev.D62:114012, hep-ph/0006304 (W,Z+bb)  *'
      write(6,*) '*  Phys.Rev.D60:113006, hep-ph/9905386 (diboson) *'
      write(6,*) '*                                                *'
      write(6,*) '* On the web:  http://mcfm.fnal.gov/             *'
      write(6,*) '**************************************************'
      write(6,*) 
 
      return
      end








